__docs__/components/gallery/component-gallery.tsx

```diff
@@ -1,49 +1,125 @@
+
     return (
-        <View>
-            <AccordionGallerySection />
-            <BannerSection />
-            <BirthdayPickerSection />
-            <BreadcrumbsSection />
-            <ButtonSection />
-            <CellSection />
-            <DropdownSection />
-            <FormSection />
-            <IconButtonSection />
-            <IconSection />
-            <LinkSection />
-            <ModalSection />
-            <PillSection />
-            <PopoverSection />
-            <ProgressSpinnerSection />
-            <SearchFieldSection />
-            <SwitchSection />
-            <ToolbarSection />
-            <TooltipSection />
-        </View>
+        <RenderStateRoot>
+            <View>
+                <Banner
+                    kind="info"
+                    layout="floating"
+                    text={`Note: The core, data, translations, layout, testing,
+                    theming, timing, tokens, and typography packages are not
+                    featured in this gallery.`}
+                />
+                <View style={styles.menuBar}>
+                    <LabelMedium>Group by</LabelMedium>
+                    <Strut size={spacing.xSmall_8} />
+                    <SingleSelect
+                        selectedValue={currentGroup}
+                        onChange={setCurrentGroup}
+                        // Placehoder is not used here
+                        placeholder=""
+                    >
+                        <OptionItem label="alphabet" value="alphabet" />
+                        <OptionItem label="function" value="function" />
+                        <OptionItem label="package" value="package" />
+                    </SingleSelect>
+                    <Strut size={spacing.large_24} />
+
+                    <LabelMedium>Layout</LabelMedium>
+                    <Strut size={spacing.xSmall_8} />
+                    <SingleSelect
+                        selectedValue={currentLayout}
+                        onChange={(newValue) =>
+                            setCurrentLayout(
+                                newValue === "grid" ? "grid" : "list",
+                            )
+                        }
+                        // Placehoder is not used here
+                        placeholder=""
+                    >
+                        <OptionItem label="grid" value="grid" />
+                        <OptionItem label="list" value="list" />
+                    </SingleSelect>
+
+                    {currentLayout === "grid" && (
+                        <>
+                            <Spring />
+                            <Switch
+                                checked={compactGridView}
+                                onChange={setCompactGridView}
+                            />
+                            <Strut size={spacing.xSmall_8} />
+                            <LabelMedium>Compact grid</LabelMedium>
```

Comment:
**suggestion:** Same suggestion here. In fact, I wonder what would happen if you use `LabelMedium` as the parent container and include the label text and the `Switch` instance in it (maybe vertical centering fails?).

Reply:
I tried this, but it looked like the Switch stopped working if I did that 🤔